packages/core/src/render3/definition.ts

```diff
@@ -741,9 +742,27 @@ export function getPipeDef<T>(type: any): PipeDef<T>|null {
   return type[NG_PIPE_DEF] || null;
 }
 
+/**
+ * Checks whether a given Component, Directive or Pipe is marked as standalone.
+ * See this guide for additional information: https://angular.io/guide/standalone-components
+ *
+ * @param type A reference to a Component, Directive or Pipe.
+ * @publicApi
+ */
 export function isStandalone<T>(type: Type<T>): boolean {
+  if (getNgModuleDef(type)) {
+    return false;
+  }
+
   const def = getComponentDef(type) || getDirectiveDef(type) || getPipeDef(type);
-  return def !== null ? def.standalone : false;
+  if (def === null) {
```

Comment:
We'd also need to add the `ngDevMode` here, so that this code is tree-shaken away from production bundles:
```suggestion
  if (ngDevMode && def === null) {
```